# Project Proposal and Literature Review

## 🔗 Builds on (AIAT 111–125)

AIAT 126 is the *graduation* project: it **assembles** what you already built across the diploma, it does not re-teach it. Open what you need:

- **AIAT 111 U2 — where a real AI problem comes from, and which techniques suit which domain** → [`Course 01/unit2-ai-concepts/examples/05_machine_learning_intro.ipynb`](../../../Course%2001/unit2-ai-concepts/examples/05_machine_learning_intro.ipynb)
- **AIAT 115 U1 — framing a question the data can actually answer** → [`Course 05/unit1-introduction/examples/01_data_science_intro.ipynb`](../../../Course%2005/unit1-introduction/examples/01_data_science_intro.ipynb)
- **AIAT 116 U5 — regulations and permitted use, for your dataset-licence line** → [`Course 06/unit5-governance-regulations/examples/01_global_regulations.ipynb`](../../../Course%2006/unit5-governance-regulations/examples/01_global_regulations.ipynb)
- **AIAT 113 U5 — reading a paper's numbers critically for the literature review** → [`Course 03/unit5-probability/examples/07_hypothesis_testing_procedures.ipynb`](../../../Course%2003/unit5-probability/examples/07_hypothesis_testing_procedures.ipynb)

Record every artifact you reuse in **section 6, the Prior Work Inventory**, of [`TEMPLATES/project_proposal_template.md`](../../TEMPLATES/project_proposal_template.md). It is scored at gate 1 (criterion 1.7) and re-checked at the Unit 2 design review — it is this course's **CLO2** evidence, and the official AIAT 126 description requires it.

## 📚 Learning Objectives

By completing this notebook, you will:
- Define clear project goals and research questions
- Conduct comprehensive literature reviews
- Write project proposals with proper structure
- Define success metrics and evaluation criteria
- Create project timelines and resource plans

## 🔗 Where this leads

**Used later in:** Course 12 — Unit 4, which measures the model against the success metrics you commit to here, and Unit 5, where the proposal becomes the report's introduction.

---

This notebook covers practical activities from **Course 12, Unit 1**:
- Selecting and defining a graduation project topic
- Conducting literature review and compiling relevant research papers
- Writing a comprehensive project proposal document
- Creating project timeline and resource allocation plan
- Defining success metrics and evaluation criteria

---

## Introduction

**Project Planning** is the foundation of a successful graduation project. It involves identifying a meaningful problem, understanding existing solutions, and planning your approach.


---

## The $62 million proposal

The **University of Texas MD Anderson Cancer Center** spent about four years building the **Oncology Expert Advisor** on IBM Watson: a system that would read a patient's record plus the oncology literature and recommend a treatment. A **48-page University of Texas System audit report** — dated November 2016, made public in early 2017 — ended it. The audit records what the money bought: at least **$62 million** spent; an original contract to deliver the leukaemia product in **six months for $2.4 million**, extended **twelve times** to total fees of **$39.2 million**; a procurement that had gone around MD Anderson's normal IT process. And the finding that matters most here — the system **was never used to treat a patient**. IBM's own agreement stated it "is not ready for human investigational or clinical use, and its use in the treatment of patients is prohibited."

Nobody on that project was incompetent, and the technology was not the problem. The failure is visible entirely in the documents this notebook teaches you to write: the objective was never narrowed to something a fixed budget could reach, *done* was never defined as a number anyone could check, and there was no point at which somebody was obliged to say **stop**.

### What goes wrong without this

A capstone with no written scope does not fail loudly in week 1. It fails quietly in week 9. You are still "improving" the model; the dataset you assumed was available turned out to need a licence; one extra feature looked cheap and cost two weeks; and there is no document anyone can point at to settle whether that feature was ever part of the project. That is **scope creep**, and its cost is always paid in the same place — the report and the demo, the last two deliverables, get compressed into whatever days are left. In AIAT 126 those two are gates 5 and 6: **30 of the 100 points**.

The proposal is the cheapest insurance in the course. It is a few hours of writing that turns *"no, that is out of scope"* into a sentence you are allowed to say — to your supervisor, to a helpful classmate with a great idea, and to yourself at 2am in week 9.

### The same reasoning, three other domains

- **Zillow Offers, November 2021** — an automated home-pricing model bought houses at prices above what Zillow could later sell them for. Q3 2021 carried a **$304.4 million** inventory write-down, the programme was wound down and roughly **25%** of the workforce went with it. The company's own explanation was that the model's error rate proved "far more volatile than we thought possible": the success criterion was never stress-tested against the market it would actually meet.
- **Google Flu Trends, 2013** — a system built to predict the CDC's influenza reports ended up predicting **more than double** the CDC's actual proportion of flu-related doctor visits (Lazer et al., *Science*, 2014). The objective was perfectly well defined; the assumption underneath it expired without anyone noticing.
- **The Netflix Prize, 2009** — a $1 million competition, won with a 10% RMSE improvement, whose winning ensemble Netflix never put into production because "the additional accuracy gains that we measured did not seem to justify the engineering effort" (Netflix, 2012). The metric was hit and the goal was still missed.

Different industries, one pattern: **the goal that was written down and the goal that mattered were not the same goal.** That gap is closed on paper, in this unit, or it is not closed at all.


## 📥 Inputs & 📤 Outputs

**Inputs:** What we use in this notebook

- Three planning templates you fill in yourself — scope, literature review, timeline — plus the
  real gate weights of AIAT 126, taken from [`PROJECT_GUIDELINES.md`](../../PROJECT_GUIDELINES.md).

**Outputs:** What you'll see when you run the cells

- The three filled templates printed, and a **plan-vs-rubric table** comparing where the
  14-week timeline spends its weeks against where the six milestone gates award their points.

---


In [1]:
# Setup: load the small toolkit this planning notebook needs — json to
# pretty-print the proposal templates, and datetime to anchor the timeline.
import json
import pandas as pd
from datetime import datetime, timedelta

# Confirm the environment is ready before filling in any template
print("✅ Libraries imported!")
print("Ready for project planning and proposal writing!")


✅ Libraries imported!
Ready for project planning and proposal writing!


## Part 1: Project Scope Definition

Define your project goals, research questions, and scope.


In [2]:
# Project scope template: write down EVERY planning decision — problem,
# objectives, research questions, scope boundaries, users, success criteria —
# before any code exists; a fuzzy scope here becomes a stalled project later.
project_scope = {
    "project_title": "Example: AI-Powered Medical Diagnosis Assistant",
    "problem_statement": "Brief description of the problem you're solving",
    # Objectives are specific and measurable — the proposal is graded against them
    "objectives": [
        "Objective 1: Develop an AI model for medical diagnosis",
        "Objective 2: Achieve >90% accuracy on validation set",
        "Objective 3: Deploy model as a web application"
    ],
    # Research questions: what the finished project should be able to answer
    "research_questions": [
        "What are the most effective ML algorithms for medical diagnosis?",
        "How can we handle class imbalance in medical datasets?",
        "What evaluation metrics are appropriate for diagnostic systems?"
    ],
    # Explicit in/out-of-scope lists keep the project finishable in one semester
    "scope": {
        "in_scope": [
            "Binary classification for specific medical conditions",
            "Using publicly available medical datasets",
            "Web-based deployment"
        ],
        "out_of_scope": [
            "Real-time diagnosis with live patient data",
            "Multi-disease classification",
            "Mobile app development"
        ]
    },
    # Naming the users forces the deliverable to serve someone concrete
    "target_users": "Healthcare professionals, medical students",
    # Numeric success criteria make "done" testable instead of a feeling
    "success_criteria": {
        "accuracy": "> 90%",
        "recall": "> 85% (critical for medical diagnosis)",
        "deployment": "Web application accessible and functional"
    }
}

# Pretty-print the filled template so you can copy this structure for your own project
print("=" * 60)
print("Project Scope Template")
print("=" * 60)
print(json.dumps(project_scope, indent=2))

print("\n✅ Use this template to define your project scope!")
print("📝 Document your project scope clearly before proceeding")


Project Scope Template
{
  "project_title": "Example: AI-Powered Medical Diagnosis Assistant",
  "problem_statement": "Brief description of the problem you're solving",
  "objectives": [
    "Objective 1: Develop an AI model for medical diagnosis",
    "Objective 2: Achieve >90% accuracy on validation set",
    "Objective 3: Deploy model as a web application"
  ],
  "research_questions": [
    "What are the most effective ML algorithms for medical diagnosis?",
    "How can we handle class imbalance in medical datasets?",
    "What evaluation metrics are appropriate for diagnostic systems?"
  ],
  "scope": {
    "in_scope": [
      "Binary classification for specific medical conditions",
      "Using publicly available medical datasets",
      "Web-based deployment"
    ],
    "out_of_scope": [
      "Real-time diagnosis with live patient data",
      "Multi-disease classification",
      "Mobile app development"
    ]
  },
  "target_users": "Healthcare professionals, medical students",

## Part 2: Literature Review Framework

Structure for conducting and organizing literature review.


In [3]:
# Literature review template: a structured record for every paper, dataset
# and tool you survey — filling these fields per paper is what turns "I read
# some papers" into a review with a defensible gap analysis.
literature_review_structure = {
    "research_papers": [
        {
            "title": "Paper Title 1",
            "authors": "Author 1, Author 2",
            "year": 2023,
            "key_findings": "Main contributions and findings",
            "relevance": "Why this paper is relevant to your project",
            "methodology": "Approach used in the paper",
            "limitations": "What this paper doesn't address"
        }
    ],
    # Track candidate datasets alongside papers — data availability shapes the project
    "datasets_used": [
        {
            "dataset_name": "Example Dataset",
            "description": "Dataset description",
            "source": "URL or reference",
            "size": "Number of samples",
            "relevance": "Why this dataset is useful"
        }
    ],
    # State of the art + gap analysis: what already exists, and where your contribution fits
    "state_of_the_art": {
        "best_approaches": "Current best methods in the field",
        "performance_baselines": "Current state-of-the-art performance metrics",
        "gap_analysis": "What's missing and how you'll contribute"
    },
    # Record the tools each paper used — they become your implementation shortlist
    "tools_and_libraries": [
        "Library 1: Purpose and usage",
        "Library 2: Purpose and usage"
    ]
}

# Summarize the review recipe and the standard places to search for papers
print("=" * 60)
print("Literature Review Structure")
print("=" * 60)
print("\nKey Components:")
print("1. Research Papers: Review 15-25 relevant papers")
print("2. Datasets: Identify datasets used in related work")
print("3. State-of-the-Art: Understand current best methods")
print("4. Gap Analysis: Identify opportunities for contribution")
print("5. Tools & Libraries: Document relevant technologies")

print("\n📚 Sources for Literature Review:")
print("  - Google Scholar")
print("  - arXiv.org")
print("  - IEEE Xplore")
print("  - ACM Digital Library")
print("  - PubMed (for medical AI projects)")


Literature Review Structure

Key Components:
1. Research Papers: Review 15-25 relevant papers
2. Datasets: Identify datasets used in related work
3. State-of-the-Art: Understand current best methods
4. Gap Analysis: Identify opportunities for contribution
5. Tools & Libraries: Document relevant technologies

📚 Sources for Literature Review:
  - Google Scholar
  - arXiv.org
  - IEEE Xplore
  - ACM Digital Library
  - PubMed (for medical AI projects)


## Part 3: Project Timeline and Resource Planning

Create a realistic timeline for your graduation project.


In [4]:
# Project timeline template: the 14-week capstone broken into 5 milestones,
# each with tasks and CONCRETE deliverables — a deliverable is what makes a
# milestone checkable instead of a vague intention.
# Anchor the plan at today; adapt the week ranges to your own calendar
start_date = datetime.now()

timeline = {
    "Week 1-2": {
        "milestone": "Project Proposal", "tasks": [
            "Define project scope and objectives",
            "Conduct initial literature review",
            "Write project proposal document"
        ],
        "deliverables": ["Project Proposal Document"]
    },
    "Week 3-5": {
        "milestone": "Data Collection & Preparation",
        "tasks": [
            "Identify and acquire datasets",
            "Data cleaning and preprocessing",
            "Exploratory data analysis",
            "Feature engineering"
        ],
        "deliverables": ["Clean dataset", "EDA notebook"]
    },
    "Week 6-10": {
        "milestone": "Model Development",
        "tasks": [
            "Model architecture design",
            "Training and validation",
            "Hyperparameter tuning",
            "Model evaluation"
        ],
        "deliverables": ["Trained models", "Evaluation results"]
    },
    "Week 11-12": {
        "milestone": "Deployment & Testing",
        "tasks": [
            "Model deployment",
            "System testing",
            "Performance validation"
        ],
        "deliverables": ["Deployed system"]
    },
    "Week 13-14": {
        "milestone": "Documentation & Presentation",
        "tasks": [
            "Write final report",
            "Create presentation",
            "Prepare demonstration"
        ],
        "deliverables": ["Final Report", "Presentation", "Video Demo"]
    }
}

# Print each milestone with a sample of its tasks and every deliverable
print("=" * 60)
print("Project Timeline Template (14 weeks)")
print("=" * 60)
for week, details in timeline.items():
    print(f"\n{week}: {details['milestone']}")
    print(f"  Tasks: {', '.join(details['tasks'][:2])}...")
    print(f"  Deliverables: {', '.join(details['deliverables'])}")

print("\n✅ Adjust timeline based on your project complexity!")


Project Timeline Template (14 weeks)

Week 1-2: Project Proposal
  Tasks: Define project scope and objectives, Conduct initial literature review...
  Deliverables: Project Proposal Document

Week 3-5: Data Collection & Preparation
  Tasks: Identify and acquire datasets, Data cleaning and preprocessing...
  Deliverables: Clean dataset, EDA notebook

Week 6-10: Model Development
  Tasks: Model architecture design, Training and validation...
  Deliverables: Trained models, Evaluation results

Week 11-12: Deployment & Testing
  Tasks: Model deployment, System testing...
  Deliverables: Deployed system

Week 13-14: Documentation & Presentation
  Tasks: Write final report, Create presentation...
  Deliverables: Final Report, Presentation, Video Demo

✅ Adjust timeline based on your project complexity!


In [5]:
# Comparative check: does the 14-week timeline above spend its weeks where AIAT 126
# actually awards its points? WHY compare the two: a plan is only credible when the
# calendar matches the rubric, and that comparison is arithmetic, not opinion.
# The gate weights are the real ones from PROJECT_GUIDELINES.md ("the six milestone gates").
# Gate 6 (the defense) is left out on purpose: it happens in its own session AFTER the
# build calendar, so charging it zero weeks here would be misleading rather than informative.
GATE_POINTS = {
    "Gate 1 - Proposal":       10,
    "Gate 2 - Design review":  10,
    "Gate 3 - Implementation": 30,
    "Gate 4 - Evaluation":     20,
    "Gate 5 - Final report":   15,
}

# Which gate does each milestone in `timeline` feed? Keys are read from the plan above,
# so this check can never drift away from the plan it is checking.
MILESTONE_TO_GATE = {
    "Project Proposal":              "Gate 1 - Proposal",
    "Data Collection & Preparation": "Gate 3 - Implementation",
    "Model Development":             "Gate 3 - Implementation",
    "Deployment & Testing":          "Gate 4 - Evaluation",
    "Documentation & Presentation":  "Gate 5 - Final report",
}


def weeks_in(label):
    """'Week 6-10' -> 5. The plan is written as ranges; the comparison needs a count."""
    span = label.replace("Week", "").strip()
    first, _, last = span.partition("-")
    return int(last or first) - int(first) + 1


weeks_per_gate = {gate: 0 for gate in GATE_POINTS}
unmapped = []
for week_label, details in timeline.items():
    gate = MILESTONE_TO_GATE.get(details["milestone"])
    if gate is None:
        unmapped.append(details["milestone"])          # a milestone nothing grades
    else:
        weeks_per_gate[gate] += weeks_in(week_label)

total_weeks = sum(weeks_per_gate.values())
plan_vs_rubric = pd.DataFrame({
    "points": pd.Series(GATE_POINTS),
    "weeks_planned": pd.Series(weeks_per_gate),
})
plan_vs_rubric["%_of_marks"] = (
    plan_vs_rubric["points"] / plan_vs_rubric["points"].sum() * 100).round(1)
plan_vs_rubric["%_of_calendar"] = (
    plan_vs_rubric["weeks_planned"] / total_weeks * 100).round(1)
plan_vs_rubric["gap_pp"] = (
    plan_vs_rubric["%_of_calendar"] - plan_vs_rubric["%_of_marks"]).round(1)

print("=" * 78)
print(f"PLAN vs RUBRIC - {total_weeks} planned weeks against "
      f"{plan_vs_rubric['points'].sum()} points (gates 1-5)")
print("=" * 78)
print(plan_vs_rubric.to_string())

# The two findings a reviewer would draw from this table, computed rather than asserted
starved = plan_vs_rubric[plan_vs_rubric["weeks_planned"] == 0]
print(f"\nGates with NO milestone anywhere in the timeline: "
      f"{list(starved.index) if len(starved) else 'none'} "
      f"({int(starved['points'].sum())} points)")
print(f"Milestones that feed no gate: {unmapped if unmapped else 'none'}")
print(f"Biggest over-spend  : {plan_vs_rubric['gap_pp'].idxmax():<24} "
      f"{plan_vs_rubric['gap_pp'].max():+.1f} percentage points of calendar vs marks")
print(f"Biggest under-spend : {plan_vs_rubric['gap_pp'].idxmin():<24} "
      f"{plan_vs_rubric['gap_pp'].min():+.1f} percentage points of calendar vs marks")

PLAN vs RUBRIC - 14 planned weeks against 85 points (gates 1-5)
                         points  weeks_planned  %_of_marks  %_of_calendar  gap_pp
Gate 1 - Proposal            10              2        11.8           14.3     2.5
Gate 2 - Design review       10              0        11.8            0.0   -11.8
Gate 3 - Implementation      30              8        35.3           57.1    21.8
Gate 4 - Evaluation          20              2        23.5           14.3    -9.2
Gate 5 - Final report        15              2        17.6           14.3    -3.3

Gates with NO milestone anywhere in the timeline: ['Gate 2 - Design review'] (10 points)
Milestones that feed no gate: none
Biggest over-spend  : Gate 3 - Implementation  +21.8 percentage points of calendar vs marks
Biggest under-spend : Gate 2 - Design review   -11.8 percentage points of calendar vs marks


### Reading the comparison: the template you just printed has a hole in it

The numbers above are not a criticism of you — they are a criticism of the generic 14-week plan in the previous cell, which is the plan most students copy. Two things the table says, and nothing more:

1. **Gate 2 (the design review, 10 points) gets zero weeks.** The timeline jumps straight from "Project Proposal" to "Data Collection & Preparation". But gate 2 is not just 10 points, it is a **hard stop**: `PROJECT_GUIDELINES.md` says implementation does not start until the design review passes. A calendar that allocates no time to it is a calendar that plans to fail a gate, or to skip it and rebuild later.
2. **Implementation takes 57.1% of the calendar for 35.3% of the marks** (+21.8 percentage points), while evaluation takes 14.3% of the calendar for 23.5% of the marks (−9.2). The build feels like the project; the rubric says the build is barely a third of it.

Neither number tells you the plan is *wrong*. Building genuinely does take longer than evaluating. What the table gives you is the sentence a supervisor will ask for: **"I am spending 57% of my weeks on 35% of my marks, deliberately, because X."** If you cannot finish that sentence, move a week.

**Fix your own copy before you submit it:** add a design-review milestone between the proposal and the data work, and re-run this cell.

## 💬 Discuss

No single right answer here — argue your position and say what evidence would change it.

1. **Is the plan wrong, or is the rubric?** The table shows 57.1% of the calendar going to a gate worth 35.3% of the marks. Defend one of two positions to your instructor: *(a)* the 14-week template is right and building really does deserve most of the calendar, so the rubric under-weights it; or *(b)* the template is wrong and those weeks belong to design and evaluation. Whichever you pick — **what would you have to measure in your own project to find out you were mistaken?**
2. **When the data says your target was fantasy.** The worked scope above commits to "accuracy > 90%" and "recall > 85%" for a medical diagnosis assistant — typed before anyone opened a file. Suppose in week 4 your data supports 78% recall and no more. Three responses: lower the target, change the problem, or report the miss and keep the target. Who is helped and who is harmed by each? MD Anderson's project kept an ambitious target for four years and treated nobody — does that argue for or against lowering targets early?
3. **Which of your out-of-scope items are choices, and which are laws?** The template lists "real-time diagnosis with live patient data" as out of scope. In Saudi Arabia, patient records are personal data under the **Personal Data Protection Law**, so for a clinical project that line may be a legal boundary rather than a scoping preference. Go through your own out-of-scope list and mark each item **choice** or **constraint**. What has to change in your proposal — in the objectives, the dataset section, or the risks section — when an item moves from one column to the other?

## ⚠️ Where this breaks

A proposal is a commitment made at the moment you know least about the project. The templates above are worth writing anyway — but here is where they stop helping.

- **A proposal cannot be tested, so it cannot be wrong yet.** Every number in `success_criteria` was typed before anyone looked at the data. This is exactly why Unit 2 makes you compute a **baseline on the real file**: in that lesson a one-line domain rule ("predict survived if female") beats *both* trained models on F1, which would have exposed an over-confident proposal target in an afternoon. The assumption that must hold for your proposal numbers to mean anything: **you have seen enough of the data to know its ceiling.** Until then, treat every target as provisional and say so in writing. Do not defend a proposal number after the data has contradicted it — amend it, and record why.
- **The 14-week timeline assumes the dataset works.** There is no milestone for *"the data turned out to be unusable"*, which is the most common reason a capstone gets re-planned. The plan quietly assumes you can obtain, licence and load your data during weeks 3–5. Test that assumption in **week 1**: downloading one file and printing `df.shape` costs an hour and can save the project. If the file will not load, the licence forbids your use, or the labels are not in it, you have eleven weeks to change course instead of two.
- **15–25 papers is a lot of reading for one semester, and a bibliography is not a review.** The cheaper alternative when time is short: a **narrow** review — three papers that define the task, two that give you a baseline number you must beat, one dataset paper — read properly. Depth on six beats a skim of twenty-five, and only the depth survives a defense question.
- **When NOT to write a long proposal at all.** If your project is a reuse-and-extend of something you already built in AIAT 111–125 (your Prior Work Inventory names it), a one-page proposal plus a working `load → baseline → metric` script is stronger evidence than ten pages of plan. Planning is what you do instead of knowing; running code is knowing.
- **A plan that is never amended is not a good plan, it is an unread one.** Every departure from this proposal is legitimate if it is written down with a reason. A silent departure is scored as scope drift at gate 3.

## Summary

### Key Components of Project Proposal:
1. **Project Title**: Clear, descriptive title
2. **Problem Statement**: What problem are you solving?
3. **Objectives**: Specific, measurable goals
4. **Research Questions**: Key questions your project answers
5. **Scope**: What's included and excluded
6. **Literature Review**: Relevant research and state-of-the-art
7. **Methodology**: Technical approach and methods
8. **Timeline**: Realistic schedule with milestones
9. **Success Criteria**: How you'll measure success
10. **Risk Assessment**: Potential challenges and mitigation

### Evaluation Metrics:
- **Classification**: Accuracy, Precision, Recall, F1-score, ROC-AUC
- **Regression**: MSE, RMSE, MAE, R²
- **Deployment**: Response time, uptime, user satisfaction

**Reference:** Course 12, Unit 1: "Project Planning and Proposal" - All practical activities covered


## 📚 References

1. Kitchenham, B. (2004). *Procedures for Performing Systematic Reviews*. Keele University Technical Report TR/SE-0401.
2. Keshav, S. (2007). *How to Read a Paper*. ACM SIGCOMM Computer Communication Review, 37(3), 83–84. <https://doi.org/10.1145/1273445.1273458>
3. Page, M. J., et al. (2021). *The PRISMA 2020 statement: an updated guideline for reporting systematic reviews*. BMJ, 372:n71. <https://doi.org/10.1136/bmj.n71>

**Cases cited in this lesson**

4. University of Texas System, Audit Office. (2016). *UTMDACC Special Review of Procurement Procedures Related to Oncology Expert Advisor Project*. <https://www.utsystem.edu/documents/docs/financial/utmdacc-special-review-procurement-procedures-related-oncology-expert> — the audit that ended the MD Anderson / IBM Watson project; source of the $62 million total, the twelve contract extensions and the finding that the system was never used in patient care.
5. Zillow Group. (2021). *Q3 2021 results and plan to wind down Zillow Offers*, 2 November 2021 (Form 8-K). <https://www.sec.gov/Archives/edgar/data/1617640/000161764021000085/z-20211102.htm> — the $304.4 million inventory write-down and ~25% workforce reduction that followed a home-pricing model whose error rate was "far more volatile than we thought possible"; the reference case for a success criterion that was never stress-tested.
6. Lazer, D., Kennedy, R., King, G., & Vespignani, A. (2014). *The Parable of Google Flu: Traps in Big Data Analysis*. **Science, 343(6176), 1203–1205.** <https://doi.org/10.1126/science.1248506> — the documented case of a well-specified objective outliving the assumption it rested on.
7. Amatriain, X., & Basilico, J. (2012). *Netflix Recommendations: Beyond the 5 Stars (Part 1)*. Netflix Technology Blog — the statement that the Netflix Prize winning ensemble was not deployed because the accuracy gains "did not seem to justify the engineering effort needed to bring them into a production environment."
